src/hivpy/sexual_behaviour.py

```diff
@@ -105,20 +121,14 @@ def update_sex_groups(self, population: pd.DataFrame):
                 if any(index):
                     subpop_size = sum(index)
                     rands = np.random.uniform(0.0, 1.0, subpop_size)
-                    ages = population.loc[index, "age"]
+                    self.calc_rred_age(population, index)
+                    rred = population.loc[index, "rred"] * population.loc[index, "rred_age"]
                     dim = self.sex_behaviour_trans[sex].shape[0]
                     Pmin = np.zeros(subpop_size)
                     Pmax = np.zeros(subpop_size)
                     for new_group in range(dim):
                         Pmin = Pmax.copy()
-                        Pmax += self.prob_transition(sex, ages, prev_group, new_group)
+                        Pmax += self.prob_transition(sex, rred, prev_group, new_group)
                         # This has to be a Series so it can be combined with index correctly
-                        jump_to_new_group = pd.Series((rands >= Pmin) & (rands < Pmax),
-                                                      index=ages.index)
+                        jump_to_new_group = pd.Series((rands >= Pmin) & (rands < Pmax))
```

Comment:
Double checking that it's safe to remove the index from this Series here. Will it always be combined with `index` correctly? I remember there was  a reason for adding it but maybe it's no longer relevant in this version.